angr/knowledge_plugins/variables/variable_manager.py

```diff
@@ -640,5 +641,29 @@ def get_variable_accesses(self, variable: SimVariable, same_name: bool=False) ->
     def copy(self):
         raise NotImplementedError
 
+    @staticmethod
+    def convert_variable_list(start:int, vlist: List[Variable], manager: VariableManagerInternal ):
+        for v in vlist:
+            simv = None
+            if v.sort == "global":
+                simv = SimMemoryVariable(v.addr,v.type.byte_size)
+            elif v.sort == "register":
+                simv = SimRegisterVariable(v.addr,v.type.byte_size)
+            elif v.sort == "stack":
+                simv = SimStackVariable(v.addr, v.type.byte_size)
+            else:
+                l.warning("undefined variable sort %s for %s", v.sort, v.addr)
+                continue
+            manager.add_variable(v.sort, start, simv)
+
+    def load_from_dwarf(self, cu: CompilationUnit = None):
+        cu = cu or self._kb._project.loader.elfcore_object.compilation_units
```

Comment:
I have a hard time believing this code is written correctly - it looks like you're taking a list of compilation units and treating it as a single unit.